### PR TITLE
Update protobuf rules for examples

### PR DESCRIPTION
**Pull request by pcj**
* Remove pubref/rules_protobuf
* Add stackb/rules_proto
* Update rules_go to 0.16.0

### CODE REVIEW

**Comment by pcj** [comment]
I've tested the individual server/client example implementation but did not run a full e2e test suite (hoping buildkite will do that for me).  If passes, this PR will supercede #195 as this one updates rules_go to the desired version

cc @katre

**Comment by nlopezgi** [comment]
thanks a lot for putting this together @pcj !
looks like //examples/hellohttp:deployment_test is failing, could you double check running it locally?
also kicked off buildkite tests that will allow us to get the full error log

**Comment by pcj** [comment]
/retest

All tests passing on my ubuntu 17 laptop

```
$ bazel test //...
INFO: Build completed successfully, 84 total actions
```

**Comment by katre** [comment]
Thanks very much, @pcj !

**Comment by nlopezgi** [comment]
/retest 
@erain I'm not sure the failure is related to the line
`Error from server (NotFound): deployments.apps "hello-grpc-staging" not found`
From what I could tell, this line is also present in passing runs:
https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/bazelbuild_rules_k8s/200/pull-rules-k8s-e2e/301/?log#log+
https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/bazelbuild_rules_k8s/193/pull-rules-k8s-e2e/299/?log#log

I dug a bit more in the log and the relevant error message is:
`./examples/hellogrpc/e2e-test.sh: line 47: ./bazel-bin/examples/hellogrpc/go/client/client: No such file or directory
`
It looks like the new `/examples/hellogrpc/go/client:client` target is not producing the expected `./bazel-bin/examples/hellogrpc/go/client/client` output, but instead its producing as output `./bazel-bin/examples/hellogrpc/go/client/linux_amd64_stripped/client` (probably due to changes in rules_go).
The issue is the script expects the target to have that name here (https://github.com/bazelbuild/rules_k8s/blob/master/examples/hellogrpc/e2e-test.sh#L47).

Note that for the cc, java and py hellogrpc samples the output is indeed the expected  `./bazel-bin/examples/hellogrpc/${LANGUAGE}/client/client` so changing that line for:
`OUTPUT=$(./bazel-bin/examples/hellogrpc/${LANGUAGE}/client/linux_amd64_stripped/client $(get_lb_ip))`
does not work as it breaks execution with cc, java and py (but succeeds for go)

Should be feasible to have some hack in the script to add the `linux_amd64_stripped` only if language is go.

**Comment by pcj** [comment]
Thanks @nlopezgi for looking into this; that makes sense.  The workaround is similar to https://github.com/bazelbuild/rules_go/issues/1239#issuecomment-361296415 that I'll address.

**Comment by nlopezgi** [comment]
thanks @pcj , the workaround proposed there should work fine here. I'll kick off buildkite once you update your commit and do then a full review (also please address the comments made by @erain). Thanks again.

**Comment by nlopezgi** [comment]
this is still failing, 
looks like the normalized rule is producing
bazel-genfiles/examples/hellogrpc/go/client/client
but the script is looking for
./bazel-bin/examples/hellogrpc/go/client/client

File: WORKSPACE [hunk 1]
```diff
@@ -119,27 +118,57 @@ py_library(
...
 # Imports for examples/
 # ================================================================
 
-git_repository(
-    name = "org_pubref_rules_protobuf",
-    commit = "4cc90ab2b9f4d829b9706221d4167bc7fb3bd247",  # patched v0.8.1 (Sep 27 2017)
-    remote = "https://github.com/pubref/rules_protobuf.git",
+http_archive(
+    name = "build_stack_rules_proto",
+    urls = ["https://github.com/stackb/rules_proto/archive/32204c6ec15aea9c015774223252be09d4156234.tar.gz"],
```
[the remark this answers is not in the document]
Comment (pcj) [reply WORKSPACE]: The previous rules were written before the `native.proto_library` rule was stable.  As a consequence they were somewhat incompatible with other rules in the bazel proto ecosystem and perhaps more complex than necessary. These rules are simpler and a lot less code to maintain.  I'll update the PR message and reconsider that word.

**Comment by pcj** [comment]
You are too fast and too right sir!  Added `output_to_bindir`.  In resolving @erain comment I took the liberty of updating to rules_go 0.16.1.  PR squashed to a single commit and force pushed.

**Comment by nlopezgi** [comment]
hi @pcj ,
we've fixed some of the flakyness issues with our e2e tests if you want to try to update this branch so we can get it in. Let me know if you dont think you'll have the time, if so we'll try to find someone else that can pick up and finish it.
Thanks!

**Comment by nlopezgi** [comment]
Hi @pcj ,
Im going to try to fix this PR and get it merged as its blocking https://github.com/bazelbuild/rules_k8s/issues/227

**Comment by nlopezgi** [comment]
@pjc, thanks again for putting this pr together

**Pull request merged**